frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -164,6 +186,29 @@ const StrategyFormBody = ({
     }
   }, [selectedStrategyType, values, setFieldValue]);
 
+  useEffect(
+    function removeDefaultValuesFromForm() {
+      if (isSubmitting) {
+        for (const fieldName in values) {
+          const value = values[fieldName as keyof CacheStrategy];
+          const defaultValue = getDefaultValueForField(values.type, fieldName);
+          if (
+            defaultValue !== undefined &&
+            defaultValue !== null &&
+            (value === defaultValue ||
+              // Formik initially records numbers as strings so we need to
+              // compare them as numbers
+              Number(value) === Number(defaultValue))
```

Comment:
I'm not sure I'm following the exact need for conversion here. My first comment would be to prefer comparison by strings if possible, second would be to make sure you want to use `Number` over `parseInt`/`parseFloat`.